cfgov/unprocessed/js/modules/Tree.js

```diff
@@ -0,0 +1,122 @@
+'use strict';
+
+/**
+ * Tree
+ * @class
+ *
+ * @classdesc A tree data structure.
+ * Tree's have one root node, and child nodes that branch.
+ * Like:
+ *
+ *        R
+ *       / \
+ *      A   B
+ *    / | \
+ *   C  D  E
+ *
+ * @returns {Tree} An instance.
+ */
+function Tree() {
+
+  var _root = null;
+  var _levelCache = {};
+
+  /**
+   * @param {Object} payload - Data to attach to the root node.
+   * @returns {Tree} An instance.
+   */
+  function init( payload ) {
+    _root = new TreeNode( this, payload );
+    _levelCache[0] = [ _root ];
+
+    return this;
+  }
+
+  /**
+   * @param {Object} payload - Data to attach to new node.
+   * @param {TreeNode} parent - Node on which to add a new child node.
+   * @returns {TreeNode} The newly instantiated and added node.
+   */
+  function add( payload, parent ) {
+    if ( !parent.children ) { parent.children = []; }
```

Comment:
Why is this check necessary? Doesn't the parent `TreeNode` always have children? https://github.com/cfpb/cfgov-refresh/pull/1542/files#diff-6ea501203f930030236f6ca28adaf8feR116